service/src/main/java/com/commercetools/pspadapter/payone/mapping/MappingUtil.java

```diff
@@ -127,25 +128,41 @@ public static void mapCustomerToRequest(@Nonnull final AuthorizationRequest requ
                 });
     }
 
-    public static void mapShippingAddressToRequest(final AuthorizationRequest request, final Address shippingAddress) {
-
-        if(shippingAddress == null) {
+    public static void mapShippingAddressToRequest(final AuthorizationRequest request,
+                                                   final Address shippingAddress,
+                                                   final String paymentMethod) {
+
+        if(shippingAddress != null) {
+            request.setShipping_firstname(shippingAddress.getFirstName());
+            request.setShipping_lastname(shippingAddress.getLastName());
+            request.setShipping_street(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getStreetName(),
+                shippingAddress.getStreetNumber())));
+            request.setShipping_zip(shippingAddress.getPostalCode());
+            request.setShipping_city(shippingAddress.getCity());
+            request.setShipping_country(shippingAddress.getCountry().toLocale().getCountry());
+            request.setShipping_company(joinStringsIgnoringNull(Arrays.asList(shippingAddress.getCompany(),
+                shippingAddress.getDepartment())));
+
+            if (countriesWithStateAllowed.contains(shippingAddress.getCountry())) {
+                request.setShipping_state(shippingAddress.getState());
+            }
+        } else if(!MethodKeys.WALLET_PAYPAL.equals(paymentMethod)) {
             throw new IllegalArgumentException("Missing shipping address details");
```

Comment:
Is shippingAddress required for all other payment methods?

Reply:
Good question.
Same like Paypal payments we have to check all other Payment methods documentation for understanding how to process payments for digital items.
At least this is not part of this task.